packages/devtools_app/lib/src/performance/timeline_flame_chart.dart

```diff
@@ -1036,7 +1036,12 @@ class SelectedFrameBracketPainter extends FlameChartPainter {
 
   @override
   void paint(Canvas canvas, Size size) {
-    if (selectedFrame == null) return;
+    if (selectedFrame == null) {
+      canvas
```

Comment:
why this save and restore? that should be a no-op.

Reply:
you cant call `restore` on Canvas without first calling `save`. It throws an exception if you do

Reply:
right but why are you doing a save + restore with nothing in the middle?
That seem like writing:
```dart
foo += 1;
foo -= 1;
```

Reply:
this is to ensure that the selected frame painter (the blue brackets that surround the frame's timeline events) disappears when a frame is unselected. We essentially need to repaint an empty scene for this custom painter.

Reply:
hmm - maybe this was a reloading issue (or operator error) when I saw this behavior earlier. Was able to remove the canvas calls and still get the desired effect.